server/tso/tso.go

```diff
@@ -56,8 +56,11 @@ type timestampOracle struct {
 	// TODO: remove saveInterval
 	saveInterval  time.Duration
 	maxResetTSGap func() time.Duration
-	tso           unsafe.Pointer
-	lastSavedTime atomic.Value
+	// tso info stored in the memory
+	tsoMux sync.RWMutex
+	tso    *tsoObject
+	// last timestamp window stored in etcd
+	lastSavedTime atomic.Value // stored as time.Time
```

Comment:
What about:

``` golang
type timestampOracle struct {
    ...
    tsoMux sturct {
       sync.RwMutex
       tso *tsoObject
    }
    ...
}
```

This is more clear to show tso should be under tsoMu controlled.